controllers/datadogagent/component/utils.go

```diff
@@ -478,3 +487,504 @@ func ShouldCreateAgentLocalService(gitVersion string, forceEnableLocalService bo
 	// Service Internal Traffic Policy exists in Kube 1.21 but it is enabled by default since 1.22
 	return utils.IsAboveMinVersion(gitVersion, localServiceDefaultMinimumVersion) || (utils.IsAboveMinVersion(gitVersion, localServiceMinimumVersion) && forceEnableLocalService)
 }
+
+// BuildCiliumPolicy creates the base node agent, DCA, or CCR cilium network policy
+func BuildCiliumPolicy(dda metav1.Object, site string, ddURL string, hostNetwork bool, dnsSelectorEndpoints []metav1.LabelSelector, componentName v2alpha1.ComponentName) (string, string, []cilium.NetworkPolicySpec) {
+	policyName, podSelector := GetNetworkPolicyMetadata(dda, componentName)
+	var policySpecs []cilium.NetworkPolicySpec
+
+	switch componentName {
+	case v2alpha1.NodeAgentComponentName:
+		policySpecs = []cilium.NetworkPolicySpec{
+			egressECSPorts(podSelector),
+			egressNTP(podSelector),
+			egressMetadataServerRule(podSelector),
+			egressDNS(podSelector, dnsSelectorEndpoints),
+			egressAgentDatadogIntake(podSelector, site, ddURL),
+			egressKubelet(podSelector),
+			ingressDogstatsd(podSelector),
+			egressChecks(podSelector),
+		}
+	case v2alpha1.ClusterAgentComponentName:
+		_, nodeAgentPodSelector := GetNetworkPolicyMetadata(dda, v2alpha1.NodeAgentComponentName)
+		policySpecs = []cilium.NetworkPolicySpec{
```

Comment:
Similar to the above, I think we're missing an ingress rule when cluster runners are enabled, and another one when external metrics are enabled.